lib/node_modules/@stdlib/string/base/replace-after/lib/main.js

```diff
@@ -0,0 +1,98 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2023 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
+var isInteger = require( '@stdlib/assert/is-integer' ).isPrimitive;
+var format = require( '@stdlib/string/format' );
+
+
+// MAIN //
+
+/**
+* Replaces the substring after the first occurrence of a specified search string.
+*
+* @param {string} str - input string
+* @param {string} replacement - replacement string
+* @param {string} search - search string
+* @param {integer} [fromIndex=0] - index at which to start the search
+* @throws {TypeError} first argument must be a string
+* @throws {TypeError} second argument must be a string
+* @throws {TypeError} third argument must be a string
+* @throws {TypeError} fourth argument must be an integer
+* @returns {string} substring
+*
+* @example
+* var out = replaceAfter( 'Hello, world!', 'foo', ', ' );
+* // returns 'Hello, foo'
+*
+* @example
+* var out = replaceAfter( 'beep boop', 'foo', 'beep' );
+* // returns 'beepfoo'
+*
+* @example
+* var out = replaceAfter( 'beep boop', 'foo', 'boop' );
+* // returns 'beep boopfoo'
+*
+* @example
+* var out = replaceAfter( 'beep boop', 'foo', 'xyz' );
+* // returns ''
+*
+* @example
+* var out = replaceAfter( 'beep boop', 'foo', 'beep', 5 );
+* // returns ''
+*
+* @example
+* var out = replaceAfter( 'beep boop beep baz', 'foo', 'beep', 5 );
+* // returns 'beep boop beepfoo'
+*/
+function replaceAfter( str, replacement, search, fromIndex ) {
+	var idx;
+	if ( !isString( str ) ) {
```

Comment:
You can remove input argument validation as this is a "base" package where we assume we've been provided valid inputs.

Reply:
Should I remove the validation of the 4th argument "fromIndex" too?

Reply:
Yeah, you can remove the `isInteger` check.